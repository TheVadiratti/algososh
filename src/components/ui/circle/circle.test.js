import renderer from 'react-test-renderer';
import { Circle } from './circle';

describe('Circle component tests', () => {

  it('circle renders without symbol', () => {
    const button = renderer.create(<Circle />).toJSON();
    expect(button).toMatchSnapshot();
  })

  it('circle renders with symbol', () => {
    const button = renderer.create(<Circle letter='A' />).toJSON();
    expect(button).toMatchSnapshot();
  })

  it('circle renders with head', () => {
    const button = renderer.create(<Circle head='HEAD' />).toJSON();
    expect(button).toMatchSnapshot();
  })

  it('circle renders with react-component head', () => {
    const button = renderer.create(<Circle head={<p>TEST</p>} />).toJSON();
    expect(button).toMatchSnapshot();
  })

  it('circle renders with tail', () => {
    const button = renderer.create(<Circle tail='HEAD' />).toJSON();
    expect(button).toMatchSnapshot();
  })

  it('circle renders with react-component tail', () => {
    const button = renderer.create(<Circle tail={<p>TEST</p>} />).toJSON();
    expect(button).toMatchSnapshot();
  })

  it('circle renders with index', () => {
    const button = renderer.create(<Circle index={1} />).toJSON();
    expect(button).toMatchSnapshot();
  })

  it('circle renders isSmall', () => {
    const button = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(button).toMatchSnapshot();
  })

  it('circle renders default', () => {
    const button = renderer.create(<Circle state='default' />).toJSON();
    expect(button).toMatchSnapshot();
  })

  it('circle renders changing', () => {
    const button = renderer.create(<Circle state='changing' />).toJSON();
    expect(button).toMatchSnapshot();
  })

  it('circle renders modified', () => {
    const button = renderer.create(<Circle state='modified' />).toJSON();
    expect(button).toMatchSnapshot();
  })
})