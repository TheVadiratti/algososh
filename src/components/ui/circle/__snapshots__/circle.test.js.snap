// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component tests circle renders changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle renders default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle renders isSmall 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle renders modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle renders with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    HEAD
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle renders with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle component tests circle renders with react-component head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <p>
      TEST
    </p>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle renders with react-component tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <p>
      TEST
    </p>
  </div>
</div>
`;

exports[`Circle component tests circle renders with symbol 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle renders with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    HEAD
  </div>
</div>
`;

exports[`Circle component tests circle renders without symbol 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
